packages/amazon-vpc-cni-plugins/amazon-vpc-cni-plugins.spec

```diff
@@ -0,0 +1,115 @@
+
+# Construct reproducible tar archives
+# See https://reproducible-builds.org/docs/archives/
+%global source_date_epoch 1234567890
+%global tar_cf tar --sort=name --mtime="@%{source_date_epoch}" --owner=0 --group=0 --numeric-owner -cf
+
+Name: %{_cross_os}amazon-vpc-cni-plugins
+Version: %{vpccni_gover}
+Release: 1%{?dist}
+Summary: VPC CNI plugins for Amazon ECS and EKS
+License: Apache-2.0
+URL: https://%{vpccni_goimport}
+Source0: https://%{vpccni_goimport}/archive/%{vpccni_gitrev}/%{vpccni_gorepo}.tar.gz
+
+BuildRequires: %{_cross_os}glibc-devel
+
+Requires: %{name}(binaries)
+
+%description
+%{summary}.
+
+%package bin
+Summary: VPC networking plugins binaries
+Provides: %{name}(binaries)
+Requires: (%{_cross_os}image-feature(no-fips) and %{name})
+Conflicts: (%{_cross_os}image-feature(fips) or %{name}-fips-bin)
+
+%description bin
+%{summary}.
+
+%package fips-bin
+Summary: VPC networking plugins binaries, FIPS edition
+Provides: %{name}(binaries)
+Requires: (%{_cross_os}image-feature(fips) and %{name})
+Conflicts: (%{_cross_os}image-feature(no-fips) or %{name}-bin)
+
+%description fips-bin
+%{summary}.
+
+%prep
+%autosetup -n %{vpccni_gorepo}-%{vpccni_gitrev}
+
+# Symlink amazon-vpc-cni-plugins-%{vpccni_gitrev} to the GOPATH location
+%cross_go_setup %{vpccni_gorepo}-%{vpccni_gitrev} %{vpccni_goproject} %{vpccni_goimport}
+
+%build
+# cross_go_configure cd's to the correct GOPATH location
+%cross_go_configure %{vpccni_goimport}
+LD_VPC_CNI_VERSION="-X github.com/aws/amazon-vpc-cni-plugins/version.Version=%{vpccni_gover}"
+VPC_CNI_HASH="%{vpccni_gitrev}"
+LD_VPC_CNI_SHORT_HASH="-X github.com/aws/amazon-vpc-cni-plugins/version.GitShortHash=${VPC_CNI_HASH::8}"
+
+declare -a VPC_CNI_BUILD_ARGS
+VPC_CNI_BUILD_ARGS=(
+  -ldflags "${GOLDFLAGS} ${LD_VPC_CNI_VERSION} ${LD_VPC_CNI_SHORT_HASH} ${LD_VPC_CNI_PORCELAIN}"
+)
+
+for p in \
+  aws-appmesh \
+  ecs-serviceconnect \
+  vpc-branch-eni \
+  vpc-bridge \
+  vpc-eni \
+  vpc-tunnel \
+; do
+  go build "${VPC_CNI_BUILD_ARGS[@]}" -mod=vendor -o ${p} ./plugins/${p}
+  gofips build "${VPC_CNI_BUILD_ARGS[@]}" -mod=vendor -o fips/${p} ./plugins/${p}
+done
+
+%install
+install -d %{buildroot}%{_cross_libexecdir}
+install -D -p -m 0755 aws-appmesh %{buildroot}%{_cross_libexecdir}/cni/vpc/aws-appmesh
+install -D -p -m 0755 ecs-serviceconnect %{buildroot}%{_cross_libexecdir}/cni/vpc/ecs-serviceconnect
+install -D -p -m 0755 vpc-branch-eni %{buildroot}%{_cross_libexecdir}/cni/vpc/vpc-branch-eni
+install -D -p -m 0755 vpc-bridge %{buildroot}%{_cross_libexecdir}/cni/vpc/vpc-bridge
+install -D -p -m 0755 vpc-eni %{buildroot}%{_cross_libexecdir}/cni/vpc/vpc-eni
+install -D -p -m 0755 vpc-tunnel %{buildroot}%{_cross_libexecdir}/cni/vpc/vpc-tunnel
+
+install -d %{buildroot}%{_cross_fips_libexecdir}
+install -D -p -m 0755 fips/aws-appmesh %{buildroot}%{_cross_fips_libexecdir}/cni/vpc/aws-appmesh
+install -D -p -m 0755 fips/ecs-serviceconnect %{buildroot}%{_cross_fips_libexecdir}/cni/vpc/ecs-serviceconnect
```

Comment:
@arnaldo2792 does adding these new plugins change the ECS agent's behavior? Service Connect still won't be supported - will the node register with that feature only to fail task launches that use it, or will the plugin be ignored?

Reply:
The plugin will be ignored, the ECS agent looks for a Service Connect container image:


https://github.com/aws/amazon-ecs-agent/blob/4e8af8ac81a9e7dfafd96964eda8c1ff3c88f478/agent/app/agent_capability.go#L481